src/github/auto-upgrade-dependencies.ts

```diff
@@ -0,0 +1,143 @@
+import { Component } from '../component';
+import { NodeProject } from '../node-project';
+
+/**
+ * Options for `AutoUpgradeDependencies`.
+ */
+export interface AutoUpgradeDependenciesOptions {
+
+  /**
+   * Schedule to run on.
+   *
+   * @default AutoUpgradeDependenciesSchedule.DAILY
+   */
+  readonly schedule?: AutoUpgradeDependenciesSchedule;
+
+  /**
+   * Auto approve PR's, allowing mergify to merge them.
+   *
+   * Note that if the project doesn't have the auto-approve workflow configured,
+   * enabling this will throw an error, as auto approval cannot be accomplished.
+   *
+   * @default - true if auto-approve workflow is configured for the project, false otherwise.
+   */
+  readonly autoApprove?: boolean;
+
+  /**
+   * List of package names to ignore during the upgrade.
+   *
+   * @default - Only projen itself is ignored.
```

Comment:
```suggestion
   * @default - `projen` is ignored since there is a separate workflow to upgrade it.
```

Which makes me wonder - with the combination of this and the self-mutation feature, do we really need a separate upgrade workflow for projen? Feel like we can just use this workflow to upgrade projen itself, and let the self-mutation workflow update the branch if there are any synth changes, no?

Reply:
tough question.